# Interpretation (Nestle1904LFT)

## Table of content <a class="anchor" id="TOC"></a>
* <a href="#bullet1">1 - Introduction</a>
* <a href="#bullet2">2 - Load Text-Fabric app and data</a>
* <a href="#bullet3">3 - Examine some cases</a>
     * <a href="#bullet3x1">3.1 - Louw Nida and John 1:1 (Ἐν ἀρχῇ)</a>
     * <a href="#bullet3x2">3.2 - Formal and functional tags (ἔρχεται)</a>
     * <a href="#bullet3x3">3.3 - Handling of multiple appositions</a>

# 1 - Introduction <a class="anchor" id="bullet1"></a>
##### [Back to TOC](#TOC)

This Jupyter Notebook shows a number of cases where the interpretation of the Greek text involves some kind of choice that had to be made. However, it does not necessarily claim the data is 'wrong'.

# 2 - Load Text-Fabric app and data <a class="anchor" id="bullet2"></a>
##### [Back to TOC](#TOC)

For this notebook two data sets of Text-Fabric will be loaded. As consequence the option `hoist=globals()` can not be used and the Advanced API calls require to be called in a differnt manner. 

In [41]:
%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [42]:
# Loading the Text-Fabric code
# Note: it is assumed Text-Fabric is installed in your environment
from tf.fabric import Fabric
from tf.app import use

In [46]:
# load the N1904GBI app and data
N1904GBI = use ("tonyjurg/Nestle1904GBI", version="0.3")

**Locating corpus resources ...**

The requested data is not available offline
	~/text-fabric-data/github/tonyjurg/Nestle1904GBI/tf/0.3 not found


In [44]:
# load the N1904LFT app and data
N1904LFT = use ("tonyjurg/Nestle1904LFT", version="0.3")

**Locating corpus resources ...**

The requested app is not available offline
	~/text-fabric-data/github/tonyjurg/Nestle1904LFT/app not found


findAppClass: invalid syntax (~/text-fabric-data/github/tonyjurg/Nestle1904LFT/app/app.py, line 5)


findAppClass: Api for "tonyjurg/Nestle1904LFT" not loaded
The requested data is not available offline
	~/text-fabric-data/github/tonyjurg/Nestle1904LFT/tf/0.3 not found


In [47]:
# The following will push the Text-Fabric stylesheet to this notebook (to facilitate proper display with notebook viewer)
# The stylesheets of N1904GBI and N1904LFT will be identical, so one call will suffice.
N1904GBI.dh(N1904.getCss())

# 3 - Performing the queries <a class="anchor" id="bullet3"></a>
##### [Back to TOC](#TOC)

## 3.1 - Louw Nida and John 1:1 (Ἐν ἀρχῇ)<a class="anchor" id="bullet3x1"></a>
##### [Back to TOC](#TOC)

Examin the Greek ἀρχῇ in John 1:1:

In [82]:
WordQuery = '''
word book_long=John chapter=1 verse=1 word=ἀρχῇ
'''
WordResult = N1904LFT.search(WordQuery) 
# returns list of ordered tuples, even when only containing one node, hence indexed with [0][0] on next line
print("word:",T.text(WordResult[0][0])," Louw-Nida:",F.ln.v(WordResult[0][0]))

  0.10s 1 result
word: ἀρχῇ   Louw-Nida: 67.65


However, the ἀρχή of John 1:1, according to Louw-Nida Lexicon can be either:
>```
a:beginning (aspect)=68.1  [^1]
b:beginning (time)=67.65   [^2]

Only the second meaning is provided by the feature 'ln'. Although the most appropriate, the meaning of the second one (aspect) might also be implied by the author. Queries depending on 'ln' (or, by consequence, 'lex_dom') may yield incorrect or incomplete results.

[^1] Louw, Johannes P., and Eugene Albert Nida. *Greek-English Lexicon of the New Testament: Based on Semantic Domains, Vol. I* (New York: United Bible Societies, 1996), 654.

[^2] ibid., 636.

## 3.2 - Formal and functional tags (ἔρχεται)<a class="anchor" id="bullet3x2"></a>
##### [Back to TOC](#TOC)

Functional tags are only available on GBI, so the data will be taken from that Text-Fabric dataset. See [this Jupyter notebook](https://github.com/tonyjurg/Nestle1904GBI/blob/main/docs/usecases/formal_versus_functional_tag.ipynb) for an indepth discussion about formal and functional tags.

Consider the two following verses, both containing the same verb ἔρχεται:
> Καὶ ἐξῆλθεν ἐκεῖθεν, καὶ **ἔρχεται** εἰς τὴν πατρίδα αὐτοῦ, .. (He went away from there and **came** to his hometown, Mark 6:1a, ESV) 

> Ἰδοὺ **ἔρχεται** μετὰ τῶν νεφελῶν, καὶ ὄψεται αὐτὸν πᾶς ὀφθαλμὸς ... (Behold, **he is coming** with the clouds, and every eye will see him, Rev. 1:7a ESV)

The following query pulls the relevant data from the GBI Text-Fabric dataset

In [84]:
ErgetaiQuery = '''
word word=ἔρχεται
/with/
book=Mark chapter=6 verse=1 
/or/
book=Revelation chapter=1 verse=7 
/-/
'''

ErgetaiResult = N1904GBI.search(ErgetaiQuery) 
# returns list of ordered tuples

# We also need to add the app reference (N1904GBI) before we can access the F API functions
api=N1904GBI.api
for tuple in ErgetaiResult:
    node=tuple[0]
    print(T.sectionFromNode(node), " word:",T.text(node)," Formal tag:",api.F.formaltag.v(node),
      " Functional tag ",api.F.functionaltag.v(node)," gloss:",api.F.gloss_EN.v(node))

  0.10s 2 results
('Mark', 6, 1)  word: ἔρχεται   Formal tag: V-PNI-3S  Functional tag  V-PNI-3S  gloss: came
('Revelation', 1, 7)  word: ἔρχεται   Formal tag: V-PNI-3S  Functional tag  V-PNI-3S  gloss: He is coming


Both instances are the same praesens indicativus of ἔρχομαι, with identical formal and functional tag, but require to be translated rather different. 

## 3.3 - Handling of multiple appositions<a class="anchor" id="bullet3x3"></a>
##### [Back to TOC](#TOC)

Consider the following text from Matthew 1:1:

> Βίβλος γενέσεως Ἰησοῦ Χριστοῦ υἱοῦ Δαυεὶδ υἱοῦ Ἀβραάμ. 

In this verse there are two appositions to 'Ἰησοῦ Χριστοῦ': 'υἱοῦ Δαυεὶδ' and 'υἱοῦ Ἀβραάμ'. The LFT Text-Fabric data presents 'υἱοῦ Δαυεὶδ' as apposition to 'Ἰησοῦ Χριστοῦ' and 'υἱοῦ Ἀβραάμ' as apposition to 'Ἰησοῦ Χριστοῦ υἱοῦ Δαυεὶδ'. Another choice of apposition can also be argued: both 'υἱοῦ Δαυεὶδ' and 'υἱοῦ Ἀβραάμ' being appositions to 'Ἰησοῦ Χριστοῦ'. This actualy the case in the XML data for the GBI nodes.

This Example is discussed in detail in a separate [Jupiter NoteBook](https://github.com/tonyjurg/Nestle1904LFT/blob/main/docs/usecases/appostions.ipynb)

## 3.4 Adverbial to what?<a class="anchor" id="bullet3x4"></a>
##### [Back to TOC](#TOC)

To what is the κατὰ in 1 Cor. 16:2 the adverbial? The Text-Farbic data suggest to 'μίαν σαββάτου'.

In [102]:
# create the query template
VerseQuery = '''
word verse=2 chapter=16 bookshort=1Cor
'''
# execute the query template
VerseResult = N1904LFT.search(VerseQuery) 

  0.09s 20 results


In [101]:
N1904LFT.show(VerseResult, condensed=True, multiFeatures=False)